### PR TITLE
Import Misp Json format

### PR DESCRIPTION
**Pull request by sebdraven**

### CODE REVIEW

File: core/common/misp_to_yeti.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        
+        obs_yeti.add_context('misp',context)
+
+    def obs_misp_to_yeti(self, object_misp:dict):
+        objs_type = object_misp.get("type")
+        links = []
+        for attr in object_misp.get("Attribute"):
+            obs_yeti = self.attr_misp_to_yeti(attr)
+            links.append(obs_yeti)     
+        obs_yeti = links.pop()
```
Comment (tomchop) [inline core/common/misp_to_yeti.py:50]: Why pop() here? Why is the first item of the loop the one that gets links to everything else?

File: core/common/misp_to_yeti.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        for obj_to_link in links:
+            obs_yeti.link_to(obj_to_link,f'link_by_misp_{objs_type}','misp')
```
Comment (tomchop) [inline core/common/misp_to_yeti.py:52]: link_to takes 3 params: the target object, a short description (will show in the link table), and a longer description (will only show on hover; so you can add more context there)

File: tests/apiv2/import_data.py [hunk 3]
```diff
@@ -0,0 +1,2513 @@
...
+        token_data = client.post(
+            "/api/v2/auth/api-token", headers={"x-yeti-apikey": user.api_key}
+        ).json()
+        client.headers = {"Authorization": "Bearer " + token_data["access_token"]}
+
+        cls.data = {
+    "Event": {
+        "Attribute": [
+            {
+                "Galaxy": [],
```
Comment (tomchop) [inline tests/apiv2/import_data.py:33]: Let's put all this blob in a file under `tests/misp_test_data/misp.json`

File: tests/apiv2/import_data.py [hunk 4]
```diff
@@ -0,0 +1,2513 @@
...
+            json.dump(cls.data,misp_event)
+
+    def test_import_misp(self):
+        logging.info('Test import misp')
+        with open(self.path_json, 'rb') as fichier:
+            files = {'misp_file_json': (self.path_json, fichier)}
+
+            r=client.post('/api/v2/import_data/import_misp_json',files=files)
+            self.assertEqual(r.status_code,200)
+
```
Comment (tomchop) [inline tests/apiv2/import_data.py:2508]: It woudl also be good to assert that the observables that are created are properly formed (right values, right neighbors, etc.)

**Review by tomchop — CHANGES_REQUESTED** [review]

**Review by udgover — CHANGES_REQUESTED** [review]
That's a good start but some questions to answer. We should also agree on the formatter / linter to use because lots of changes comes from formatting and review is harder.

Finally, you MUST add the authentication check for the new API endpoint.

File: core/schemas/observable.py [hunk 5]
```diff
@@ -187,6 +189,6 @@ def find_type(value: str) -> ObservableType | None:
 from core.schemas.observables import (asn, bic, certificate, cidr, command_line,
                                       docker_image, email, file,
                                       generic_observable, hostname, iban, imphash,
-                                      ipv4, ipv6, mac_address, md5, path,
+                                      ipv4, ipv6, mac_address, md5, path,mutex,named_pipe,
```
[the remark this answers is not in the document]
Comment (sebdraven) [reply core/schemas/observable.py:192]: It's just to add for this PR. I'll do that for the next one

File: core/web/apiv2/import_data.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import json
+from fastapi import APIRouter, Depends, HTTPException, status, UploadFile, File
+import pymisp
+from core.common.misp_to_yeti import MispToYeti
+
+router = APIRouter()
+
+@router.post("/import_misp_json", tags=["import_misp_json"])
```
[the remark this answers is not in the document]
Comment (sebdraven) [reply core/web/apiv2/import_data.py:8]: nop, after we want to bulk csv, or stix format ;)

**Review by tomchop — APPROVED** [review]

**Review by tomchop — APPROVED** [review]

**Pull request merged**